### ISSUE TITLE
Health monitor doesn't allow system startup with failed lidar

### ISSUE THREAD

**Issue by TonysCousin** [usdot-fhwa-stol/carma-platform#865]
### Types of Issue

- [x] Anomaly report (something appears to not work correctly)
- [ ] Enhancement request (describe the enhancement being requested)
- [ ] Other (please ensure the description clarifies why the issue doesn’t fall into either of the above categories)

### Descriptive summary

When testing the white truck, I started Carma once and the UI displayed a continuous stream of caution messages saying one lidar was not functioning.  While it was great to get that info, the UI was stuck on the message, "waiting for system ready status" (exact wording may be off).  It appears that the health monitor noticed the non-functional lidar and correctly issued the caution (since GPS and the second lidar were still fine), but doing so prevented it from recognizing that it had enough driver functionality to proceed with automated operations.

If the system started normally and began driving an automated route, then one lidar failed, health monitor would send the same caution message, the UI would display it, and everything would continue operating.  Startup should be handled the same way.  I don't buy the argument that at startup it is more important to prevent operation with one sensor down.  There is just as much chance of a problem (crash, etc) regardless of where it happens in the route.  Rather than have the software presume to know what's best for the operator, it should inform the operator, then give them the flexibility to do what they think is best.  

Therefore, health monitor should not consider lidars or GPS to be essential drivers (individually).  Rather it already has logic coded to handle a variety of permutations on driver availability.  Let's depend on that and not overly simplify the situation by listing them all as essential in the config file.  The only truly essential driver, by itself, is the controller driver.

### Carma version where this issue was discovered

Version 3.3.0 + develop branch

### Expected behavior
Allow automation to be engaged with one lidar (out of multiple) non-functional.

### Actual behavior

Automation could not be engaged until all sensors are fully functional.

### Steps to reproduce the actual behavior

1. Start the electronics in a truck with multiple lidars
2. Power off one of the lidars (I did not do this intentionally, I experienced an honest system fault one time)
3. Start the carma software per normal procedures.

### Related work

Nothing current.

**Comment by msmcconnell** [comment]
During estimation, the team agreed this issue actually comprises 3 separate issues:
1. The publication of system alerts should be event driven not frequency based. See #908
2. Sensor failure behavior should be the same both at startup and during normal operation. This issue #865
3. health_monitor should not use parameters to define the exact names of required drivers. Instead this should be determined dynamically based on the driver types available. See #909

The following represents the new Description of this Issue:
## Description (Updated)

When testing the white truck, it appears that the health monitor noticed a non-functional lidar and correctly issued the caution (since GPS and the second lidar were still fine), but doing so prevented it from recognizing that it had enough driver functionality to proceed with automated operations.

If the system started normally and began driving an automated route, then one lidar failed, health monitor would send the same caution message, the UI would display it, and everything would continue operating. Startup should be handled the same way. I don't buy the argument that at startup it is more important to prevent operation with one sensor down. There is just as much chance of a problem (crash, etc) regardless of where it happens in the route. Rather than have the software presume to know what's best for the operator, it should inform the operator, then give them the flexibility to do what they think is best.

**TLDR**: The impact of a sensor failure should be consistent regardless of health_monitor state. So if one Lidar failure results in a CAUTION message, this should not prevent a DRIVERS_READY message from being published since the system can still operation.
